hibernate-core/src/main/java/org/hibernate/cfg/AvailableSettings.java

```diff
@@ -755,8 +755,26 @@
 	 */
 	String HBM2DLL_CREATE_NAMESPACES = "hibernate.hbm2dll.create_namespaces";
 
+	/**
+	 * Used to specify the {@link org.hibernate.tool.schema.spi.SchemaFilter} to use for creating the database schema.
+	 * 
+	 * @since 5.1
+	 */
+	String SCHEMA_CREATE_FILTER = "hibernate.schema.create.filter";
 
+	/**
+	 * Used to specify the {@link org.hibernate.tool.schema.spi.SchemaFilter} to use for dropping the database schema.
+	 * 
+	 * @since 5.1
+	 */
+	String SCHEMA_DROP_FILTER = "hibernate.schema.drop.filter";
 
+	/**
+	 * Used to specify the {@link org.hibernate.tool.schema.spi.SchemaFilter} to use for migrating the database schema.
+	 * 
+	 * @since 5.1
+	 */
+	String SCHEMA_MIGRATE_FILTER = "hibernate.schema.migrate.filter";
```

Comment:
If we need support for varied filtering for each I think I'd prefer one contract (and one setting) that gives access to the appropriate filter.  Something like:

````
interface SchemaFilterProvider {
    SchemaFilter getCreateFilter();
    SchemaFilter getDropFilter();
    SchemaFilter getMigrateFilter();
    SchemaFilter getValidateFilter();
}
````